Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -0,0 +1,238 @@
+package de.rki.coronawarnapp.risk
+
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationCompat
+import com.google.android.gms.nearby.exposurenotification.ExposureSummary
+import de.rki.coronawarnapp.CoronaWarnApplication
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.exception.RiskLevelCalculationException
+import de.rki.coronawarnapp.nearby.InternalExposureNotificationClient
+import de.rki.coronawarnapp.notification.NotificationHelper
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_INITIAL
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_OUTDATED_RESULTS
+import de.rki.coronawarnapp.server.protocols.ApplicationConfigurationOuterClass
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.storage.RiskLevelRepository
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.millisecondsToHours
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.math.round
+
+@Singleton
+class DefaultRiskLevels @Inject constructor(
+    private val appConfigProvider: AppConfigProvider
+) : RiskLevels {
+
+    companion object {
+        private var TAG = DefaultRiskLevels::class.simpleName
+        private const val DECIMAL_MULTIPLIER = 100
+    }
+
+    override fun updateRepository(riskLevel: RiskLevel, time: Long) {
+        val rollbackItems = mutableListOf<RollbackItem>()
+        try {
+            Timber.tag(TAG).v("update the risk level with $riskLevel")
```

Comment:
As a general thing: I'd really like it if log statements start with a capital letter, their are sentences after all.